diff --git a/usr/share/jolla-settings/pages/sailfishos/mainpage.qml b/usr/share/jolla-settings/pages/sailfishos/mainpage.qml
index ca5c814..c330236 100644
--- a/usr/share/jolla-settings/pages/sailfishos/mainpage.qml
+++ b/usr/share/jolla-settings/pages/sailfishos/mainpage.qml
@@ -82,6 +82,22 @@ Page {
         }
     }
 
+    DBusInterface {
+        id: patchmanagerBus
+        busType: DBusInterface.SystemBus
+        destination: "org.SfietKonstantin.patchmanager"
+        path: "/org/SfietKonstantin/patchmanager"
+        iface: "org.SfietKonstantin.patchmanager"
+    }
+
+    function unapplyAllCallback(arg) {
+        if (arg) {
+            storeIf.installDownloadedUpdate()
+        } else {
+            patchmanagerBus.call("applyPatch", ["sailfishos-patchmanager-unapplyall"])
+        }
+    }
+
     SilicaFlickable {
         anchors.fill: parent
         contentHeight: content.height
@@ -217,7 +217,7 @@ Page {
                             }
                         }
                     } else if (storeIf.mayInstall) {
-                        storeIf.installDownloadedUpdate()
+                        patchmanagerBus.call("unapplyAllPatches", [], unapplyAllCallback)
                     }
                 }
             }
