--- original/FlickableImageView.qml	2014-03-25 11:29:15.000000000 +0100
+++ patched/FlickableImageView.qml	2014-03-25 12:43:29.344398797 +0100
@@ -161,11 +161,11 @@
                 duration: model.duration
 
                 onClicked: {
-                   if (mediaPlayer.playbackState == MediaPlayer.PlayingState) {
-                       mediaPlayer.pause()
-                   } else {
+                   //if (mediaPlayer.playbackState == MediaPlayer.PlayingState) {
+                   //    mediaPlayer.pause()
+                   //} else {
                        view.clicked()
-                   }
+                   //}
                }
             }
         }
